ide/httpserver/src/org/netbeans/modules/httpserver/HttpServerURLMapper.java

```diff
@@ -82,7 +90,12 @@ private URL decodeURL(String path) {
                 newPath.append(tok);
             } else {
                 try {
-                    newPath.append(URLDecoder.decode(tok, "UTF-8")); // NOI18N
+                    String decodedToken = URLDecoder.decode(tok, "UTF-8"); // NOI18N
+                    if(decodedToken.startsWith("/") && decodedToken.endsWith("/")) { // NOI18N
+                        newPath.append(decodedToken.substring(1, decodedToken.length() - 1));
```

Comment:
who about
newPath.append(decodedToken, 1, decodedTocken.length() b -1) ;

it avoids sub-string allocation and a loop
turning newPath into StringBuilder  in line 86 would also improve performance a bit.